backend-rust/migrations/0001_initialize.up.sql

```diff
@@ -312,7 +312,27 @@ CREATE TABLE bakers(
     -- Fraction of finalization rewards charged by the pool owner.
     -- Stored as a fraction of an amount with a precision of `1/100_000`.
     finalization_commission
+        BIGINT,
+    -- Transaction used for self suspending.
+    -- This is not null only when a baker have send the transaction for self suspending.
+    -- This should always be null, when `inactive_suspended` or `primed_for_suspension` column is
+    -- not null.
+    self_suspended
         BIGINT
+        REFERENCES transactions,
+    -- Block which suspended this baker due to inactivity.
+    -- This is not null only when a baker got suspended due to inactivity.
+    -- This should always be null, when `self_suspended` or `primed_for_suspension` column is not
+    -- null.
+    inactive_suspended
+        BIGINT
+        REFERENCES blocks,
+    -- Block which is primed for suspension this baker due to inactivity.
+    -- This is not null only when a baker got primed for suspension due to inactivity.
+    -- This should always be null, when `self_suspended` or `inactive_suspended` column is not null.
+    primed_for_suspension
```

Comment:
Since we don't have any front-end design it is a bit unclear what makes sense to collect in the database. I see two options:
- We replicate the same data as the node captures (I think that is what Thomas suggests)
- We capture data as we want to display it at the front-end in the database, e.g. I suspect we do not overload the state of the baker at the front-end. A baker can only be displayed suspended on-time and the latest suspension event is the one that counts:

    ```
NotSuspended
Suspended(blockheight)
PrimedForSuspension(blockheight)
    ```

```
enum SuspensionCause {
            SelfSuspended(blockheight)
          InactivitySuspended( blockheight)
}
```

    ```

Reply:
Since the node tracks state for self-suspension and primed for suspensions separately, we have to do the same.
If a validator is primed for suspension, then self-suspends and resume, they would still be primed. With the above state we would not know to show them as primed in this case.